src/library/scala/concurrent/impl/Promise.scala

```diff
@@ -86,6 +86,34 @@ private[concurrent] final object Promise {
       override def apply(ignored: Try[T]): Unit = releaseShared(1)
     }
 
+    private final class Link[T](to: DefaultPromise[T]) extends AtomicReference[DefaultPromise[T]](to) {
```

Comment:
@richdoughtery I'm experimenting with making Link a first-class construct (so I can make a DefaultPromise implementation which in itself is a Callback—a TransformationPromise (not included in this commit)). The reason for having the Link reified is that I wouldn't otherwise be able to discern between a DefaultPromise which is a link, and a DefaultPromise which is a transformation. Thoughts?